variants: Document the `image-layout` options (#1896)
bottlerocket-os/bottlerocket: variants/README.md

Variants should almost always include the `release` package.
This pulls in the other core packages and includes essential configuration and services.

> This variant includes the (optional) `image-layout` section, which allows us to customize the layout of image we are building.

Reviewer: ```suggestion
This variant includes the (optional) `image-layout` section, which allows the user to customize the layout of the image they are building.
```

Or:

```suggestion
This variant includes the (optional) `image-layout` section, which allows to customize the layout of the image they are building.
```

Author: Good catch! Fixed. :)

`os-image-size-gib` is the size of the "OS" disk image in GiB.
`data-image-size-gib` is the size of the "data" disk image in GiB.
Though we've done so here for sake of demonstration, resizing the "data" disk image isn't necessary as it expands to fill the disk on boot.
`partition-plan` is the strategy used for image partitioning, with the options being "split" (the default) or "unified".
The "split" partition strategy has separate volumes for "OS" and "data", while "unified" has "OS" and "data" on a single volume.

> See the documentation for the [defaults](https://github.com/bottlerocket-os/bottlerocket/blob/4cb75aea3715de3ec91cf54d9a4699e11b5f3c24/tools/buildsys/src/manifest.rs#L260) and [additional details](https://github.com/bottlerocket-os/bottlerocket/blob/4cb75aea3715de3ec91cf54d9a4699e11b5f3c24/tools/buildsys/src/manifest.rs#L76).

Reviewer 2: This should be a markdown link instead of a GitHub "permalink". I think it will always be that particular version of the file that this link goes to.
